packages/reactive-element/src/reactive-element.ts

```diff
@@ -964,6 +964,9 @@ export abstract class ReactiveElement
 
   /**
    * Synchronizes property values when attributes change.
+   *
+   * Specifically, when an attribute is set, the corresponding property is set.
+   * You should rarely need to implement this callback.
```

Comment:
Since this is not our interface, I think we should point to the MDN docs: https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#using_the_lifecycle_callbacks

I think saying that it's rarely needed to be overridden is good, but the most important thing is saying that if you do you must call super.

And not needed now, but I'd also document the method by what it does from the platform (and hopefully we'd be consistent with the other CE callback's docs). Something like "Called whenever one of the element's attributes is changed. This is one of the standard custom element lifecycle callbacks."

Reply:
This is super useful. I can address.

Reply:
Added these notes. PTAL! Thank you!